### PR TITLE
work around libassimp-dev bug

### PR DESCRIPTION
**Pull request by dirk-thomas**
The CMake config file provided by the current Debian package sets the variable `ASSIMP_LIBRARY_DIRS` to the value `/usr/lib/lib/x86_64-linux-gnu`. The duplicate `lib` comes from the way the value is being parameterized when building the Debian package. Anyway the correct value is `/usr/lib/x86_64-linux-gnu`.

While investigating this problem I noticed that the code trying to find the library (see https://github.com/ros2/rviz/commit/826bd0b8f2487051f4768cd48e6e2d4f1993ce28#diff-5e9bf0e7ab4d201db275acc2fcd12886R28) doesn't check the result of the `find_library` call. That should happen independent of this workaround. @wjwwood Can you follow up on this when you have a chance?

@j-rivero Can you try to get this fixed upstream when you have a chance?

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
Thanks @dirk-thomas for the suggested fix.

For context to connect the issues together:
- The issue is tracked at https://github.com/ros2/rviz/issues/248. The problem affects most cmake variables and not only the library dirs. Full issue description [here](https://github.com/assimp/assimp/issues/1914)
- Our fix has been merged upstream https://github.com/assimp/assimp/pull/1915
- And requested to pull fhe fix in debian at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=896835

As we didn't hear back from the maintainer, @nuclearsandwich is planning to check with @j-rivero the next steps to get the fix released.

**Comment by dirk-thomas** [comment]
This patch simply let's me build the code in the current state without having to wait for any upstream fixes. Without it I simply can't build it successfully on Bionic since the manifest declares a dependency on the "broken" `libassimp-dev` package which is installed by `rosdep`.

**Comment by mikaelarguedas** [comment]
> This patch simply let's me build the code in the current state without having to wait for any upstream fixes. 

Yeah and that's great I think the workaround should be merged :+1:

I pointed out the additional links so that people external to the issue and tagged here knew what issue and fix we were referring to.

**Comment by mikaelarguedas** [comment]
I tried this patch but it doesn't compile for me.

If pushed https://github.com/ros2/rviz/pull/288/commits/62100980037830c43b5406450de3b8fe2b2e2678 to fix `ASSIMP_INCLUDE_DIRS` as well.

CI:
Current master: [Build Status](http://ci.ros2.org/job/ci_linux/4577/)
With the original patch: [Build Status](http://ci.ros2.org/job/ci_linux/4578/)
With https://github.com/ros2/rviz/pull/288/commits/62100980037830c43b5406450de3b8fe2b2e2678: [Build Status](http://ci.ros2.org/job/ci_linux/4580/)

**Review by mikaelarguedas — APPROVED** [review]
lgtm

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
Thanks for fixing it up @mikaelarguedas.

I agree the workaround is good to have until the upstream issues are fixed. The workaround until now was to uninstall assimp, which sucks if you have ROS 1 installed at the same time.

**Comment by wjwwood** [comment]
I wasn't sure if you ran full CI on these pr's yet @mikaelarguedas, so here's CI with `--packages-up-to rviz2 rviz_rendering_tests` (including https://github.com/ros2/ci/pull/185):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4581/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1488/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3764/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4664/)

**Comment by mikaelarguedas** [comment]
> I wasn't sure if you ran full CI on these pr's yet @mikaelarguedas, so here's CI with --packages-up-to rviz2 rviz_rendering_tests

I did not, I only ran bionic. Planning on taking a MacOS node offline to try the same patch with assimp 4.1.0 installed from brew (can be done after this is merged)

File: rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in
Comment (dirk-thomas) [inline rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in]: The pattern should have a trailing slash to avoid matching in undesired positions.

Comment (mikaelarguedas) [reply rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in]: Yeah I had the same feeling, unfortunately it cannot have the trailing slash as the path exported by the assimp debian doesnt have one "/usr/lib/include"
```
CMake Error in CMakeLists.txt:
  Imported target "rviz_rendering::rviz_rendering" includes non-existent path

    "/usr/lib/include"
```
Example of job failing with the trailing slash in the pattern : https://ci.ros2.org/job/ci_linux/4579/

Comment (dirk-thomas) [reply rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in]: The pattern should test for the end-of-string then?

Comment (mikaelarguedas) [reply rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in]: Looking at their code, it appears that end of string should work in all cases :+1:

https://github.com/ros2/rviz/pull/288/commits/a77a0404bd69c3bee4e0284ccee19f552c15ac9f

**Comment by mikaelarguedas** [comment]
CI with https://github.com/ros2/rviz/commit/a77a0404bd69c3bee4e0284ccee19f552c15ac9f
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4584/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1491/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3769/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4667/)

File: rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in
Comment (dirk-thomas) [reply rviz_assimp_vendor/rviz_assimp_vendor-extras.cmake.in]: :+1:

**Pull request merged**
